lib/transformer.dart

```diff
@@ -107,44 +115,234 @@
+
+
+  /// File copied from the template where the generated rules are written.
+  static const String GENERATED_RULES_FILE_NAME =
+      "generated_serialization_rules.dart";
+
+  /// URI that is imported in source files when using serialization.
+  static const String SERIALIZATION_IMPORT =
+      "package:serialization/serialization.dart";
 
-class SerializationTransformer extends Transformer {
   BarbackSettings _settings;
 
   get allowedExtensions => ".dart";
 
   SerializationTransformer.asPlugin(this._settings);
 
-  apply(Transform t) {
-    return t.readInputAsString(t.primaryInput.id).then((contents) {
-      var id = t.primaryInput.id;
-      var fileName = path.url.withoutExtension(id.path);
-      var format = _settings.configuration['format'];
-      var useLists = format == null || format == 'lists';
-      var text = generateCustomRulesFor(
-          contents,
-          listFormat: useLists,
-          libraryName: path.url.basename(fileName),
-          originalImport: path.url.basename(id.path));
-      var newId =
-          new AssetId(id.package, "${fileName}_serialization_rules.dart");
-      // Remove the leading /lib on the file name.
-      var fileNameInPackage = path.joinAll(path.split(id.path).skip(1));
-      if (_settings.mode == BarbackMode.DEBUG) {
-        t.logger.info("Generated serialization rules in $newId");
+  String classifyPrimary(AssetId id) {
+    return "allAssets";
+  }
+
+  apply(AggregateTransform transform) {
+
+    Completer completer = new Completer();
+
+    // Extracting settings.
+    bool useAnnotation = _settings.configuration['use_annotation'];
+    List<String> files = _settings.configuration['files'];
+    String rulesFileName = _settings.configuration['rules_file_name'];
+    if (rulesFileName == null) {
+      rulesFileName = GENERATED_RULES_FILE_NAME;
+    }
+    if (files == null) files = new List();
+    String format = _settings.configuration['format'];
+    bool useLists = format == null || format == 'lists';
+
+    // Create an Asset for each package file path defined in the pubspec
+    // transformer attribute.
+    List<String> packageFiles = files.where(
+        (String s) => s.startsWith("package:"));
+    List<Asset> packageAssets = new List();
+    packageFiles.forEach((String packageFile){
+      List<String> packageFilePathSplit =
+          packageFile.replaceFirst("package:", "").split("/");
+      String filePath = path.joinAll(
+          new List.from(packageFilePathSplit)..insert(0, "packages"));
+      String packageName = packageFilePathSplit.removeAt(0);
+      String barbackPath = path.joinAll(
+          new List.from(packageFilePathSplit)..insert(0, "lib"));
+      AssetId assetId = new AssetId(packageName, barbackPath);
+      packageAssets.add(new Asset.fromPath(assetId, filePath));
+    });
+
+    // Get the list of assets.
+    Future<List<Asset>> assetsFuture = transform.primaryInputs.toList();
+
+    assetsFuture.then((List<Asset> assets) {
```

Comment:
There's a lot of code in the transformer. I'd prefer it if we could have the transformer as much of a shell as possible and then the fundamental logic can be used both from a standalone binary/build time tool and from a transformer.  A lot of this seems to be related to replacing the import / figuring out which files we can replace it in. As per other comments, I think it would be better to have an existing file we can pre-populate and then that becomes less necessary.